### PR TITLE
jgrss/refine

### PR DESCRIPTION
**Pull request by jgrss**
This PR modifies the crop refinement layer.

### CODE REVIEW

File: scripts/01a_create_data.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+source /home/ubuntu/.pyenv/versions/venv.cnet/bin/activate
+
+PROJECT_PATH="/home/ubuntu/cultionet_train"
+CROP_COLUMN=""
+REPLACE_DICT=""
+
+cultionet create -p $PROJECT_PATH -gs 100 100 --crop-column $CROP_COLUMN \
```
Comment (nnguyen622) [inline scripts/01a_create_data.sh:9]: ok! So there are few added scripts. I assume this is to streamline the process of creating, predicting, etc...?

File: scripts/01b_create_predict_batch.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+PROJECT_PATH="/home/ubuntu/cultionet_train"
+CHUNKSIZE=200
+NUM_WORKERS=6
```
Comment (nnguyen622) [inline scripts/01b_create_predict_batch.sh:5]: oh nice example! 👍 
I see you increase the default chunksize and decrease num_workers

File: scripts/01a_create_data.sh (hunk 1 above)
Comment (jgrss) [reply scripts/01a_create_data.sh:9]: Correct, and to provide CLI examples.

File: scripts/01b_create_predict_batch.sh (hunk 2 above)
Comment (jgrss) [reply scripts/01b_create_predict_batch.sh:5]: These are just example script templates.

File: src/cultionet/model.py [hunk 3]
```diff
@@ -481,8 +429,63 @@ def fit(
...
+                num_processes=0,
+                accelerator=device,
+                log_every_n_steps=50,
+                profiler=profiler,
+                deterministic=False,
+                benchmark=False,
+            )
             # Calibrate the logits
             temperature_model = TemperatureScaling(
+                in_features=val_ds.num_features,
```
Comment (nnguyen622) [inline src/cultionet/model.py:488]: Ok, I see you added the argument `in_features`

File: src/cultionet/models/cultio.py [hunk 4]
```diff
@@ -23,10 +23,10 @@ def scale_min_max(
 class GeoRefinement(torch.nn.Module):
     def __init__(
         self,
+        in_features: int,
         in_channels: int = 21,
-        n_features: int = 16,
+        n_hidden: int = 32,
```
Comment (nnguyen622) [inline src/cultionet/models/cultio.py:28]: so you remove n_features argument and replace with `in_features`

File: src/cultionet/models/lightning.py [hunk 5]
```diff
@@ -359,14 +362,18 @@ def forward(
...
+    def set_true_labels(self, batch: Data) -> torch.Tensor:
         # in case of multi-class, `true_crop` = 1, 2, etc.
         true_crop = torch.where(
             (batch.y > 0) & (batch.y != self.edge_class), 1, 0
         ).long()
 
+        return true_crop
+
+    def calc_refined_loss(
+        self, batch: T.Union[Data, T.List], predictions: torch.Tensor
```
Comment (nnguyen622) [inline src/cultionet/models/lightning.py:374]: 👍

**Review by nnguyen622 — APPROVED** [review]

**Pull request merged**